subsys/settings/Kconfig

```diff
@@ -60,6 +60,7 @@ config SETTINGS_NVS
 	bool "NVS non-volatile storage support"
 	depends on NVS
 	depends on FLASH_MAP
+	depends on FLASH_MAP_DEV_NAME_IN_FLASH_AREA
```

Comment:
Are the plane is to use the device reference in nvs API?  `int nvs_init(struct nvs_fs *fs, const char *dev_name) -> int nvs_init(struct nvs_fs *fs, const struct device *dev)`

Reply:
^^ not in this commit of course.

Reply:
Yeah. Actually I have been planning to leave the original API call and add alternative nvs_init taking the device as a param; then I would replace every instance of the nvs_init call within zephyr (modules, etc) with the new API call but leave the original one available, maybe deprecate it.
It might be called directly by user and I want to leave people a chance to adjust before removing it completely.

Reply:
maybe we can move this dependency to NVS itself.

Reply:
Yes, I will do that.

Reply:
Retreating. The nvs does not need the `fa_dev_name`, the callers need it: for example nvs basic sample directly provided name of device.